### PR TITLE
method resolution fixes

### PR DESCRIPTION
**Pull request by mlangkabel**
* implemented solving array types in signature of Javassist methods
* extended method resolution logic to handling TypeVariable as actualType

### CODE REVIEW

File: java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistUtils.java [hunk 1]
```diff
@@ -150,8 +150,11 @@ static Type signatureTypeToType(SignatureAttribute.Type signatureType, TypeSolve
             }
             TypeParameterDeclaration typeParameterDeclaration = typeParameterDeclarationOpt.get();
             return new TypeVariable(typeParameterDeclaration);
+        } else if (signatureType instanceof SignatureAttribute.ArrayType) {
```
Comment (ftomassetti) [inline java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistUtils.java]: Good

File: java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/resolution/MethodResolutionLogic.java [hunk 2]
```diff
@@ -208,7 +208,21 @@ private static boolean isAssignableMatchTypeParametersMatchingQName(ReferenceTyp
                     }
                 }
             } else if (expectedParam.isReferenceType()) {
-                if (!expectedParam.equals(actualParam)) {
+                if (actualParam.isTypeVariable()) {
```
Comment (ftomassetti) [inline java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/resolution/MethodResolutionLogic.java]: Should we create an helper method for this?

Comment (mlangkabel) [reply java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/resolution/MethodResolutionLogic.java]: I'm on it :)

**Comment by ftomassetti** [comment]
Good, when tests are happy, I am happy too :)
As Travis finishes we can merge it

**Pull request merged**